src/main/java/net/openhft/chronicle/queue/impl/single/Pretoucher.java

```diff
@@ -36,130 +32,34 @@
  * <p>
  * Resources held by this object will be released when the underlying queue is closed.
  * <p>
- * Alternatively, the {@code shutdown()} method can be called to close the supplied queue and release any other resources. Invocation of the {@code
- * execute()} method after {@code shutdown()} has been called will cause an {@code IllegalStateException} to be thrown.
+ * Alternatively, the {@code close()} method can be called to close the supplied queue and release any other resources. Invocation of the {@code
+ * execute()} method after {@code close()} has been called will cause an {@code InvalidEventHandlerException} to be thrown.
  */
-@Deprecated(/* Move to Queue Enterprise in x.24 */)
-public final class Pretoucher extends AbstractCloseable {
-    static final long PRETOUCHER_PREROLL_TIME_DEFAULT_MS = 2_000L;
-    private final long PRETOUCHER_PREROLL_TIME_MS = Jvm.getLong("SingleChronicleQueueExcerpts.pretoucherPrerollTimeMs", PRETOUCHER_PREROLL_TIME_DEFAULT_MS);
-    private final SingleChronicleQueue queue;
-    private final NewChunkListener chunkListener;
-    private final IntConsumer cycleChangedListener;
-    private final boolean earlyAcquireNextCycle;
-    private final boolean canWrite;
-    private final PretoucherState pretoucherState;
-    private final TimeProvider pretouchTimeProvider;
-    private int currentCycle = Integer.MIN_VALUE;
-    private SingleChronicleQueueStore currentCycleWireStore;
-    private MappedBytes currentCycleMappedBytes;
-
+// TODO: x.25 turn into interface
+public class Pretoucher extends AbstractCloseable {
```

Comment:
Perhaps we can get rid of this class at all and use a Closeable Callable?

Reply:
Unfortunately we have to support the same class signature for one more release cycle

Reply:
I don't think we need to keep it if the feature is removed. Yes, we've marked it late, but already in 2,23